File: src/_03_BehavioralPattern/_03_05_Mediator/java/after/Restaurant.java
package _03_BehavioralPattern._03_05_Mediator.java.after;

public class Restaurant {

  private FrontDesk frontDesk;

  public Restaurant restaurant;

  public Restaurant(FrontDesk frontDesk) {
    this.frontDesk = frontDesk;
  }

  public void dinner(Integer guestId, String menu) {
    String roomNumber = this.frontDesk.getRoomNumberFor(guestId);
    System.out.println("dinner() : " + menu + " to " + roomNumber);
  }
  
}

File: src/_03_BehavioralPattern/_03_05_Mediator/Restaurant.js
/**
 * @typedef { import("./FrontDesk").default } FrontDesk
 */

export default class Restaurant {
  /** @type { FrontDesk } */
  _frontDesk;

  /** @param { FrontDesk } frontDesk */
  constructor(frontDesk) {
    this._frontDesk = frontDesk;
  }

  /**
   * 
   * @param { number } guestId 
   * @param { string } menu 
   */
  dinner(guestId, menu) {
    const roomNumber = this._frontDesk.getRoomNumber(guestId);
    console.log(`점심 ${menu} (을)를 ${roomNumber} 에 전달 !!`);
  }
}